filters/auth/grantcookie.go

```diff
@@ -12,6 +12,33 @@ import (
 	"golang.org/x/oauth2"
 )
 
+type CookieEncoder interface {
+	Create(config *OAuthConfig, host string, t *oauth2.Token) (*http.Cookie, error)
```

Comment:
I am not sure if the interface is really useful, because OAuthConfig is likely not working for oidc filters so it's only for grantflow and why should every filter invent their own?

Reply:
Interface is for users to provide pluggable implementation. I names OAuthConfig field `GrantCookieEncoder` to highlight that this is only fro grant flow.

Reply:
I understand what "interface" is, but why do we need it?
Do we have a user that wants to have its own implementation or do you think it might make sense to have as a user your own implementation?

Reply:
We discussed that this interface is here such that library users can add their own implementation.